src/turret.cpp

```diff
@@ -288,6 +295,139 @@ int turret_data::fire( player &p, const tripoint &target )
     return shots;
 }
 
+void vehicle::turrets_aim_and_fire_single()
+{
+    std::vector<std::string> option_names;
+    std::vector<vehicle_part *> options;
+
+    // Find all turrets that are ready to fire
+    for( auto &t : turrets() ) {
+        turret_data data = turret_query( *t );
+        if( data.query() == turret_data::status::ready ) {
+            option_names.push_back( t->name() );
+            options.push_back( t );
+        }
+    }
+
+    // Select one
+    if( options.empty() ) {
+        add_msg( m_warning, _( "None of the turrets are ready to fire." ) );
+        return;
+    }
+    const int idx = uilist( _( "Aim which turret?" ), option_names );
+    if( idx < 0 ) {
+        return;
+    }
+    vehicle_part *turret = options[idx];
+
+    std::vector<vehicle_part *> turrets;
+    turrets.push_back( turret );
+    turrets_aim_and_fire( turrets );
+}
+
+bool vehicle::turrets_aim_and_fire_all_manual( bool complain )
+{
+    std::vector<vehicle_part *> turrets = find_all_ready_turrets( true, false );
+
+    if( turrets.empty() ) {
+        if( complain ) {
```

Comment:
This check could be merged with the above check.

Reply:
`!complain` was intended as 'suppress message but do not change returned value', so merging that check would require another one for function result. I updated comments/names to be less confusing

Reply:
Look: `show_msg` check makes sense _only_ if `turrets` is empty. If `turrets` isn't empty, nothing happens at all, and you don't need to check for `show_msg` at all too. So instead of checking for emptiness and _then_ for `show_msg`, you could check for emptiness _and_ for `show_msg` simultaneously.

Reply:
I think you've missed `return false;` statement 2 lines below. If it wasn't there, then yeah, that would've worked.

Reply:
Look once again: your current checks are as follows

```
if( x ) {
    if( y ) {
      foo;
    }
}
```

It doesn't really matter what we have in `foo`. 
Let's check what if `x` is false: program will do nothing. That means `if( x )` check **by itself** (i.e. solitary) is useless. 
The whole 

```
if( y ) {
  foo;
}
```
check occurs **only** if `x` is true. That means we should merge both `x` and `y` checks:

```
if( x && y ) {
   foo;
}
```
In other words: show message only if `turrets` is empty **and** `show_msg` is true. If either of this conditions are false, don't do anything at all.

Reply:
No, my current checks are as follows
```c++
if( x ) {
    if( y ) {
        add_msg();
    }
    return false;
}
```
if I merge `x` and `y` it's either gonna be
```c++
if( x && y ) {
    add_msg();
    return false;
}
```
which is invalid, or
```c++
if( x && y ) {
    add_msg();
}
if( x ) {
    return false;
}
```
which is unnecessary duplication.

Reply:
Yeah, my bad. Sorry.